3rdparty/libcxx/CMakeLists.txt

```diff
@@ -1,85 +1,75 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 
-# Copy headers to collector-dir, patch, compile libcxx into intermittent library
-# place install rule
-
-# Use an external project to place the headers in the collector-dir
 include (ExternalProject)
-ExternalProject_Add(oelibcxx_cxx_includes
-    DOWNLOAD_COMMAND ""
-    CONFIGURE_COMMAND ""
-    BUILD_COMMAND ${CMAKE_COMMAND} -E copy_directory
-        ${CMAKE_CURRENT_LIST_DIR}/libcxx/include ${OE_INCDIR}/openenclave/libcxx
-        COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_LIST_DIR}/libcxx/include/__config ${OE_INCDIR}/openenclave/libcxx/__config_original
-        COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_LIST_DIR}/__config ${OE_INCDIR}/openenclave/libcxx/__config
-    INSTALL_COMMAND ""
-    )
-install (DIRECTORY ${OE_INCDIR}/openenclave/libcxx DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/openenclave/3rdparty)
+ExternalProject_Add(libcxx_includes
+  DOWNLOAD_COMMAND ""
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND ${CMAKE_COMMAND} -E copy_directory
+    ${CMAKE_CURRENT_LIST_DIR}/libcxx/include ${OE_INCDIR}/openenclave/libcxx
+    COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_LIST_DIR}/libcxx/include/__config ${OE_INCDIR}/openenclave/libcxx/__config_original
+    COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_LIST_DIR}/__config ${OE_INCDIR}/openenclave/libcxx/__config
+  INSTALL_COMMAND "")
+
+add_library(libcxx OBJECT
+  __dso_handle.cpp
+  libcxx/src/algorithm.cpp
+  libcxx/src/any.cpp
+  libcxx/src/bind.cpp
+  libcxx/src/chrono.cpp
+  libcxx/src/condition_variable.cpp
+  libcxx/src/debug.cpp
+  libcxx/src/exception.cpp
+  libcxx/src/functional.cpp
+  libcxx/src/future.cpp
+  libcxx/src/hash.cpp
+  libcxx/src/ios.cpp
+  libcxx/src/iostream.cpp
+  libcxx/src/locale.cpp
+  libcxx/src/memory.cpp
+  libcxx/src/mutex.cpp
+  libcxx/src/new.cpp
+  libcxx/src/optional.cpp
+  libcxx/src/random.cpp
+  libcxx/src/regex.cpp
+  libcxx/src/shared_mutex.cpp
+  libcxx/src/stdexcept.cpp
+  libcxx/src/string.cpp
+  libcxx/src/strstream.cpp
+  libcxx/src/system_error.cpp
+  libcxx/src/thread.cpp
+  libcxx/src/typeinfo.cpp
+  libcxx/src/utility.cpp
+  libcxx/src/valarray.cpp
+  libcxx/src/variant.cpp
+  libcxx/src/vector.cpp)
 
-# build sources
+add_dependencies(libcxx libcxx_includes)
 
-add_library(oelibcxx_cxx STATIC
-    __dso_handle.cpp
+set_target_properties(libcxx PROPERTIES
+  POSITION_INDEPENDENT_CODE ON)
```

Comment:
We may already be inheriting -fPIC from oecore via oelibc

Reply:
We seem to be inheriting it actually from `oecore`. I think we need to audit the use of `-fPIC` (we may just want to set `POSITION_INDEPENDENT_CODE` globally).

For now, let's leave this as the original libcxx build explicitly adds `-fPIC`.

Reply:
Filed #1168.